analyzers/src/SonarAnalyzer.CSharp/Rules/LoggerMembersNamesShouldComply.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class LoggerMembersNamesShouldComply : ParametrizedDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6669";
+    private const string MessageFormat = "Rename this {0} '{1}' to match the regular expression '{2}'.";
+    private const string DefaultFormat = "^_?[Ll]og(ger)?$";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat, isEnabledByDefault: false);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+        ImmutableArray.Create(Rule);
+
+    [RuleParameter("format", PropertyType.RegularExpression, "Regular expression used to check the field or property names against", DefaultFormat)]
+    public string Format { get; set; } = DefaultFormat;
+
+    private Regex NameRegex => new(Format, RegexOptions.Compiled, RegexConstants.DefaultTimeout);
```

Comment:
Yeah, very good point.
The problem is you cannot make it work like this, which creates a backing field:

```csharp
    private Regex NameRegex { get; } = new(Format, RegexOptions.Compiled, RegexConstants.DefaultTimeout);
```

...because of the `Format` reference, which is a property.

I initialize it on the `CompilationStart` action to do it only once per compilation.

Reply:
Another option would be to use (or be 😛) [Lazy<T>](https://learn.microsoft.com/en-us/dotnet/api/system.lazy-1?view=net-8.0)